### ISSUE TITLE
[Update] No Bugs!

### ISSUE THREAD

**Issue by Sierkinhane** [Sierkinhane/crnn_chinese_characters_rec#189]
更新了一下项目结构，在PyTorch1.2.0,cuda10.0下用中文数据集训练了一个Epoch，准确率达到了94%，没有出现NaN的情况。

**Comment by Sierkinhane** [comment]
有一些代码的细节可能有缺漏，望指正！

**Comment by Tony1236** [comment]
作者，你好，之前版本损失函数我看下你用的是torch.nn.CTCLoss(reduction='sum')，现在新版用的是torch.nn.CTCLoss()，这个有影响吗

**Comment by Sierkinhane** [comment]
没影响的！

**Comment by HiIcy** [comment]
作者，你好， 你之前用了个hook 去处理nan的情况，新版好像没看到，是怎么解决nan的情况的啊

**Comment by Sierkinhane** [comment]
应该是torch团队在1.2.0对ctc loss修复了，这次我只是更改了下代码结构和加了一些log

**Comment by HiIcy** [comment]
> 应该是torch团队在1.2.0对ctc loss修复了，这次我只是更改了下代码结构和加了一些log

哦哦，那我检查哈版本问题哎谢谢回复哈

**Comment by Tony1236** [comment]
> 没影响的！

ok,有个疑问，我看图片输入宽度都resize成160了，那怎么识别不定长的文字呢

**Comment by Sierkinhane** [comment]
https://github.com/Sierkinhane/crnn_chinese_characters_rec/issues/87 你可以参考这个问题

**Comment by gdxytim** [comment]
你好，请问新版_own.yaml.训练和测试目录结构。train和val图片目录就放在哪里。
谢谢。

**Comment by Sierkinhane** [comment]
仔细看readme, config 只要修改路径参数，按要求准备好labels文件

**Comment by daixiangzi** [comment]
对于非白底黑字的识别效果不是很好，请问怎么改善啊

**Comment by Sierkinhane** [comment]
从训练数据改善@daixiangzi

**Comment by zhaoguoqing12** [comment]
为什么我的loss一直在2点多不下降
![企业微信截图_15852159311553](https://user-images.githubusercontent.com/47818312/77632913-9b9a3a00-6f89-11ea-9e5d-18d9b2dc6acc.png)

![image](https://user-images.githubusercontent.com/47818312/77632929-a81e9280-6f89-11ea-9a06-f1a0c640ec4a.png)

**Comment by Sierkinhane** [comment]
将pytorch更新到1.2.0+，试一下新代码。

**Comment by zhaoguoqing12** [comment]
主要改动有什么地方嘛？ 还有如果我改为更深的cnn网络，您觉得会怎么样，会提高嘛

**Comment by Sierkinhane** [comment]
会有一定提升

**Comment by zhaoguoqing12** [comment]
> 会有一定提升
请问你有识别过英文嘛？ctc为什么会把像book这样的单词识别为bok，还有不同单词长度混合在一起识别并不好，怎么实现的不定长

**Comment by Sierkinhane** [comment]
#87 不定长，你可以参考这个问题。英文的话你可以finetune一下吧，

**Comment by daixiangzi** [comment]
问下，为啥我在自己生成的数据100w数据集上，90w用来训练，10w用来测试，但是经过15 epoch之后，训练loss几乎为0 了，但是测试上才40%准确率。。。。

**Comment by Sierkinhane** [comment]
过拟合吧@daixiangzi

**Comment by gdxytim** [comment]
你好，为什么我用笔记本和服务器训练CPU占用都达到100%，一个6G显存显卡一个11G都装好了CUDA及CUDNN.尝试改用其他GPU调用方法也不行。上一版本V2没有这问题。pytorch 1.2

**Comment by Sierkinhane** [comment]
PyTorch版本要好CUDA版本对应，不然只会在CPU调用。

**Comment by gdxytim** [comment]
是的，根据CUDA版本装PYTORCH.为什么同一环境运行上一版本的程序没有问题。

**Comment by Sierkinhane** [comment]
你测试一下你的环境能不能使用GPU, print(torch.cuda.is_available())

**Comment by gdxytim** [comment]
![图片](https://user-images.githubusercontent.com/46713006/78118137-24f1b680-7439-11ea-9df8-7cd10f2c50cb.png)
图片是上一版本的。 下面文字是这版的。可以用GPU。
 config = yaml.load(f)
=> creating output\2020-04-01-16-59\checkpoints
=> creating output\2020-04-01-16-59\log
cuda available =  True
load 56503 images!
load 7063 images!
Epoch: [0][0/1766]      Time 4.557s (4.557s)    Speed 7.0 samples/s     Data 1.327s (1.327s)    Loss 60.16076 (60.16076)

**Comment by gdxytim** [comment]
![图片](https://user-images.githubusercontent.com/46713006/78119792-861a8980-743b-11ea-88a5-1b61e4e7a89a.png)
train时CPU占用100%，val时40%。运行2Epoch 就死机了。

**Comment by Sierkinhane** [comment]
你指定一下GPUID为1，在config.yaml里面改

**Comment by gdxytim** [comment]
![图片](https://user-images.githubusercontent.com/46713006/78131216-46f53400-744d-11ea-90bd-72a2335ecbe7.png)
改了报错。

**Comment by gdxytim** [comment]
我改了getitem获取图片的语句为if .....else....有关系吗？

![图片](https://user-images.githubusercontent.com/46713006/78132586-92a8dd00-744f-11ea-87f3-32533996bacb.png)

**Comment by Sierkinhane** [comment]
这个没影响，那你试试将全部.to(device)改成.cuda()

**Comment by gdxytim** [comment]
也不行。。。
